beps/0001-notifications-system/README.md

```diff
@@ -160,89 +168,202 @@ The following backend service interfaces are added as part of this proposal.
 ```ts
 // TODO - We may want to add an additional wrapping here with interfaces for Notification and NotificationParameters
 
-interface SendNotificationRequest {
-  receivers: SignalReceivers;
+export type NotificationRecipients =
+  | {
+      type: 'entity';
+      entityRef: string | string[]; // typically a user or group entity, otherwise `spec.owner` of the entity/entities
+    }
+  | {
+      type: 'broadcast'; // all users
+    };
 
-  /* TODO - please contribute :) */
-}
+export type NotificationType = 'undone' | 'done' | 'saved';
 
-interface SendNotificationResponse {
+export type Notification = {
   id: string;
-  receivers: SignalReceivers;
+  userRef: string;
+  title: string;
+  description: string;
+  origin: string;
+  link: string;
+  created: Date;
+  saved: boolean;
+  read?: Date;
+  topic?: string;
+  updated?: Date;
+  icon?: string;
+  image?: string;
+};
 
-  /* TODO - please contribute :) */
+interface SendNotificationRequest {
+  recipients: NotificationRecipients;
+  title: string;
+  description: string;
+  link: string;
+  origin: string;
+  topic?: string;
+  icon?: string;
+  image?: string;
 }
 
 interface NotificationService {
-  sendNotification(
-    request: SendNotificationRequest,
-  ): Promise<SendNotificationResponse>;
+  sendNotification(request: SendNotificationRequest): Promise<Notification[]>;
 }
 ```
 
+Notification is always targeted to a single user unless it's a broadcast. The `notification-backend` will figure out which users will receive notification based on the `recipients` parameter.
```

Comment:
When keeping the notification immutable and maintaining metadata indendependetly (like "the notification has been read by the user) will avoid dupplication of the same records per user and will make future use-cases simple since the notifications will be automatically correlated.

These two approaches can be described as:
- sending one message to one user (the `NotificationRecipients` here is just a short-cut for emitting multiple messages)
- sending one message to all matching recipients (keeping a single message, matching metadata like the particular user)

Reply:
Yep, when it comes to database schema I think separate tables for notifications and user relation to notifications would make sense. Could even make sense to get an ER diagram included in this proposal